overlord/ifacestate/ifacemgr.go

```diff
@@ -174,6 +189,21 @@ func (m *InterfaceManager) StartUp() error {
 	if _, err := m.reloadConnections(""); err != nil {
 		return err
 	}
+
+	// check whether AppArmor prompting is supported, it is fine to do it
+	// once as toggling the feature imposes a restart of snapd
+	if assessAppArmorPrompting(m) {
+		m.useAppArmorPrompting = true
```

Comment:
to be on the safe side I would move this bit very early here, just after we have the lock if possible or as soon as possible? and then reconsult the flag to run the manager